src/CMakeLists.txt

```diff
@@ -1232,7 +1232,7 @@ if (BUILD_STATIC_LIBS)
     target_link_libraries (${HDF5_LIB_TARGET}
       PRIVATE
           "$<$<OR:$<BOOL:${HDF5_ENABLE_THREADSAFE}>,$<BOOL:${HDF5_ENABLE_SUBFILING_VFD}>,$<BOOL:${HDF5_ENABLE_MULTITHREAD}>>:Threads::Threads>"
-          "$<$<BOOL:${HDF5_ENABLE_MULTITHREAD}>:atomic>"
+          "$<$<AND:$<BOOL:${HDF5_ENABLE_MULTITHREAD}>,$<NOT:$<PLATFORM_ID:Darwin>>>:atomic>"
```

Comment:
Fine for this branch, though we should really avoid putting platform-specific logic here and handle this elsewhere. In develop HDF5 we're specifically not linking atomic in the first place, so I'm assuming we're relying on the compiler to do so as needed with -std=c11.